Singularity/Singularity/Utils/Optional.cs

```diff
@@ -8,7 +8,7 @@ namespace Singularity.Utils
     /// and Get() will return the value.
     /// </summary>
     /// <typeparam name="T">The type of this optional</typeparam>
-    internal sealed class Optional<T>
+    public sealed class Optional<T>
```

Comment:
There was an issue regarding the accessibility of the components (i think distributionmanager was more accessible than optional or something like that, could have been a method of it too).
I think it should be public anyway, because its a Util you want to use everywhere